app/models/user.rb

```diff
@@ -34,7 +34,7 @@ def pending_friends
 
   # Users who have requested to be friends
   def friend_requests
-    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+    inverse_friendships.map { |friendship| friendship unless friendship.confirmed }.compact
```

Comment:
- [ ] `friend_requests` can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby). Use the same association of `has_many - through` as listed for the `pending_friendship` and `confirm_friend` listed above.